README.md

```diff
@@ -52,13 +52,16 @@ It is recommended that you use this tool alongside the official Snyk CLI, not re
 ### Options
 
 ```console
-snyker --retries 3 --lockfile package-lock.json
+snyker --retries 3 --lockfile package-lock.json --preserve-integrity
 ```
 
-| Flag                  | Description                                                            | Default     |
-| --------------------- | ---------------------------------------------------------------------- | ----------- |
-| `--lockfile <string>` | Specify the lockfile to use (e.g. `yarn.lock` or `package-lock.json`). | `yarn.lock` |
-| `--retries <int>`     | Will set the number of times to retry logical steps of Snyker.         | `2`         |
+| Flag                   | Description                                                              | Default     |
+| ---------------------- | -------------------------------------------------------------------------| ----------- |
+| `--lockfile <string>`  | Specify the lockfile to use (e.g. `yarn.lock` or `package-lock.json`).   | Attempts to find a `yarn.lock` or `package-lock.json` then defaults to `yarn.lock` |
```

Comment:
I think we can probably auto detect.

See [here](https://github.com/asosteam/asos-core-githubactions/pull/40/files#diff-393ec533eff9feafea49f970b4c7930b823c73ef0fbbe9fcaa5119e0f1724c31) for example.

Might be worth, since we're bothering, supporting all the valid `packageManager` that node supports (i.e. `npm`, `yarn` and `pnpm`).  [Here's](https://github.com/asosteam/asos-core-githubactions/pull/40/files#diff-49b91411e60ce402922f4ad92e37a6716bd39f48ac32d839ad55a1ef0039629d) a script to detect the package manager.

Reply:
```shell
PACKAGE_MANAGER=$(
  jq '
    .packageManager // "" | 
    split("@") | 
    .[0]' $PACKAGE_PATH/package.json | 
  tr -d '"'
)
if [ "$PACKAGE_MANAGER" == "null" ]; then
  PACKAGE_MANAGER=$(
    if [ -f "$PACKAGE_PATH/yarn.lock" ] && yarn --version &> /dev/null; then
      echo yarn
    elif [ -f "$PACKAGE_PATH/pnpm-lock.yaml" ] && pnpm --version &> /dev/null; then
      echo pnpm
    else
      echo npm
    fi
  )
fi
echo $PACKAGE_MANAGER
```

Reply:
Thanks @asos-tomp, at this point in time, snyker only supports yarn and npm, other package managers currently out of scope of this change but could be a good idea for a future addition.

Reply:
Ah OK I guess I didn't appreciate the code within snyker that's specifically coupled to these two schemas.